### ISSUE TITLE
Remove old Gazelle

### ISSUE THREAD

**Issue by jayconrod** [bazelbuild/rules_go#1199]
Gazelle has moved to its own separate repository, [github.com/bazelbuild/bazel-gazelle](https://github.com/bazelbuild/bazel-gazelle). In order to avoid confusion, we'd like to remove the old copy of Gazelle in //go/tools/gazelle, which is no longer being maintained.

Old Gazelle was briefly removed in #1173, but this was rolled back in #1182 because it broke several Kubernetes repos. In particular, see kubernetes/test-infra#6075. The specific problem is in [hack/go_install_from_commit.sh](https://github.com/kubernetes/test-infra/blob/e766de76c94d45d065ac95722b9f5b8d0fe8b975/hack/go_install_from_commit.sh). This script `go get`s a package at head, cd's into the package's directory, checks out a specific old commit, then installs the package. This script breaks on the cd step if the package doesn't exist at head.

Old Gazelle can be removed in rules_go right now, but we need to ensure this won't break Kubernetes and other projects again.

If there are any open issues that should block this removal, please comment here. I'll remove old Gazelle and close this issue when there are no more blocking issues after some time has passed. I'd like to close this by the end of January.

**Comment by achew22** [comment]
It seems to me that `go get`ing a package in the rules_go ecosystem is likely a mistake on the part of the script author. The `kubernetes/test-infra` project probably should be using the same version that is pinned in rules_go.

It seems to me that just putting it in the repo suffers from the "It was on display in the bottom of a locked filing cabinet stuck in a disused lavatory with a sign on the door saying 'Beware of the Leopard.'" problem. I think sending a note out to the announce list with a deadline is probably in order, then replacing the main.go file for Gazelle with a file that does something like.

```go
import (
  "fmt"
  "os"
)

func main() {
  fmt.Fprintf(os.Stderr, `Message explaining what happened and how to fix it if you're running into this problem. This notice is a stopgap and will be removed on Feb 1, 2018`)
  os.Exit(1)
}
```

**Comment by jayconrod** [comment]
I'm composing a message to bazel-go-discuss right now actually. #1200 will put some announcements into our READMEs, too.

Printing an error message is a good idea. I'll start with a warning, probably a week or so from now.

**Comment by jayconrod** [comment]
Reminder to anyone watching this thread: Gazelle has moved to github.com/bazelbuild/bazel-gazelle. Since kubernetes/test-infra#6075 has closed, I plan to remove the old Gazelle tree early next week.

**Comment by jayconrod** [comment]
Closed with #1321
